File: sdi3-14.Cli-REST/src/uo/sdi/clients/rest/menu/items/CreateTaskMenuItem.java
package uo.sdi.clients.rest.menu.items;

import java.util.Date;

import uo.sdi.clients.rest.dto.Task;
import uo.sdi.clients.rest.restService.RestService;
import alb.util.console.Console;
import alb.util.date.DateUtil;

public class CreateTaskMenuItem extends MainMenuItem {

	public CreateTaskMenuItem(RestService restService) {
		super(restService);
	}

	@Override
	public String getMenuName() {
		return "Crear tarea";
	}

	@Override
	protected void executeInternal() {
		Task tarea = new Task();

		this.setTitle(tarea);
		this.setComments(tarea);
		this.setPlanned(tarea);

		tarea.setCreated(DateUtil.now());

		Long catId = Console.readLong("Introduzca el Id de la categoria");

		if (catId == null) {
			Console.println("Id de categoria incorrecto");
			return;
		}

		Task t = restService.createTask(tarea, catId);

		if (t == null)
			Console.println("No se ha podido crear la tarea correctamente");
		else
			Console.println("Tarea creada: " + t);
		// if (t == null)
		// Console.println("No se ha podido enviar la tarea correctamente");
		// else {
		// Console.println("La tarea se ha enviado correctamente:");
		// Console.println(t);
		// }

	}

	private void setTitle(Task tarea) {
		tarea.setTitle(Console.readString("Introduzca el titulo:"));
	}

	private void setComments(Task tarea) {
		tarea.setComments(Console.readString("Introduzca un comentario:"));
	}

	private void setPlanned(Task tarea) {
		Date fecha = null;
		boolean formatoIncorrecto = true;
		while (formatoIncorrecto) {
			int day = Console.readInt("Introduzca el dia");
			int month = Console.readInt("Introduzca el mes");
			int year = Console.readInt("Introduzca el anio");
			try {
				fecha = DateUtil.fromDdMmYyyy(day, month, year);
				formatoIncorrecto = false;
			} catch (IndexOutOfBoundsException i) {
				Console.println("El formato de la fecha no es correcto");
			}
		}
		tarea.setPlanned(fecha);
	}
}

File: sdi3-14.Cli-REST-CSharp/sdi3-14.Cli-REST-CSharp/menu/menuItems/CreateTaskMenuItem.cs
using sdi3_14.Cli_REST_CSharp.dto;
using sdi3_14.Cli_REST_CSharp.restService;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace sdi3_14.Cli_REST_CSharp.menu.menuItems
{
    class CreateTaskMenuItem : MainMenuItem
    {

        public CreateTaskMenuItem(RestService restService) : base(restService)
        {
        }


        public override String GetMenuName()
        {
            return "Crear tarea";
        }

        protected override void executeInternal()
        {
            dto.Task tarea = new dto.Task();

            this.setTitle(tarea);
            this.setComments(tarea);
            this.setPlanned(tarea);

            tarea.createdDate = new DateTime();

            long catId = getLongFromUser("Introduzca el Id de la categoria");

            dto.Task t = restService.createTask(tarea, catId);

            if (t == null)
                Console.WriteLine("No se ha podido crear la tarea");
            else
            {
                Console.WriteLine("Tarea creada satisfactoriamente");
                Console.WriteLine(t);
            }

        }

        private void setTitle(dto.Task tarea)
        {
            Console.Write("Introduzca un titulo:");
            tarea.title = Console.ReadLine();
        }

        private void setComments(dto.Task tarea)
        {
            Console.Write("Introduzca un comentario:");
            tarea.comments = Console.ReadLine();
        }

        private void setPlanned(dto.Task tarea)
        {
            DateTime fecha;
            while (true)
            {
                int day = getIntFromUser("Introduzca el dia");
                int month = getIntFromUser("Introduzca el mes");
                int year = getIntFromUser("Introduzca el anio");
                try
                {
                    fecha = new DateTime(year, month, day);
                    break;
                }
                catch (ArgumentOutOfRangeException a)
                {
                    System.Console.WriteLine("El formato de la fecha no es correcto");
                }
            }

            tarea.plannedDate = fecha;
        }


        private long getLongFromUser(String message)
        {

            long userInput = -1;

            do
            {
                System.Console.Write(message + ": ");
                String result = Console.ReadLine();

                if (long.TryParse(result, out userInput))
                    return userInput;

                System.Console.WriteLine("Formato de numero incorrecto");
            } while (true);

        }

        private int getIntFromUser(String message)
        {

            Int32 userInput = -1;

            do
            {
                System.Console.Write(message + ": ");
                String result = Console.ReadLine();

                if (Int32.TryParse(result, out userInput))
                    return userInput;

                System.Console.WriteLine("Formato de numero incorrecto");
            } while (true);

        }

    }
}
